### PR TITLE
Allow scroll suppression via shouldUpdateScroll option

### PR DESCRIPTION
**Pull request by dylanscott**
I'm not sure if you're interested in PRs on this repo (no pressure), but this could be one way of addressing #20. Happy to update this PR with documentation if this is a change you'd consider.

### CODE REVIEW

File: modules/utils/createUseScroll.js [hunk 1]
```diff
@@ -27,13 +27,18 @@ export default function createUseScroll(updateScroll, start, stop) {
...
+      let oldLocation
       let listeners = [], currentLocation, unlisten
 
       function onChange(location) {
+        oldLocation = currentLocation
         currentLocation = location
 
         listeners.forEach(listener => listener(location))
-        updateScroll(location)
+        if (!shouldUpdateScroll || shouldUpdateScroll(oldLocation, currentLocation)) {
```
Comment (taion) [inline modules/utils/createUseScroll.js]: Can you default this to a no-op function that just returns `true`? Makes the code read a bit better.

**Comment by taion** [comment]
That one nit aside, I'm okay with merging this and cutting a patch release.

I do expect the API for doing this to be considerably different post-rewrite, but this looks like a reasonable stopgap. Thanks!

**Comment by dylanscott** [comment]
Awesome, thank you!

**Comment by nikcorg** [comment]
@dylanscott you are now my personal hero! This is exactly what I needed.

**Comment by travisbloom** [comment]
@dylanscott this is really helpful, thanks!

**Comment by camsong** [comment]
@dylanscott you save my day, thanks

**Pull request merged**